site/src/main/java/uk/nhs/digital/common/components/FacetComponent.java

```diff
@@ -3,20 +3,45 @@
 import org.hippoecm.hst.core.component.HstRequest;
 import org.hippoecm.hst.core.component.HstResponse;
 import org.hippoecm.hst.site.HstServices;
+import org.onehippo.taxonomy.api.Category;
 import org.onehippo.taxonomy.api.Taxonomy;
 import org.onehippo.taxonomy.api.TaxonomyManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import uk.nhs.digital.ps.beans.HippoBeanHelper;
 
+import java.util.Locale;
+
 public class FacetComponent extends SearchComponent {
+    private static Logger log = LoggerFactory.getLogger(FacetComponent.class);
 
     @Override
     public void doBeforeRender(HstRequest request, HstResponse response) {
         TaxonomyManager taxonomyManager = HstServices.getComponentManager().getComponent(TaxonomyManager.class.getName());
         Taxonomy taxonomy = taxonomyManager.getTaxonomies().getTaxonomy(HippoBeanHelper.getTaxonomyName());
+        TaxonomyWrapper taxonomyWrapper = new TaxonomyWrapper(taxonomy);
 
-        request.setAttribute("taxonomy", taxonomy);
+        request.setAttribute("taxonomy", taxonomyWrapper);
         request.setAttribute("query", getQueryParameter(request));
         request.setAttribute("facets", getFacetNavigationBean(request));
         request.setAttribute("cparam", getComponentInfo(request)) ;
     }
+
+    public class TaxonomyWrapper {
+        private final Taxonomy taxonomy;
+
+        private TaxonomyWrapper(Taxonomy taxonomy) {
+            this.taxonomy = taxonomy;
+        }
+
+        public String getValueName(String key) {
+            Category taxonomyCategory = taxonomy.getCategoryByKey(key);
+            if (taxonomyCategory == null) {
+                log.error("No taxonomy for key: " + key);
+                return "Unknown Option: " + key;
```

Comment:
Something to check with PO. 
In terms of presentation, maybe it's better to return nothing, or do some guessing like replace "-" with " " etc...

Reply:
Have updated after input from Karen, happy for it not be be pretty as it should only happen with bad data on test/uat and shouldn't ever happen on prod.